modular_skyrat/master_files/code/modules/mob/living/emote_popup.dm

```diff
@@ -6,29 +6,62 @@
 	appearance_flags = APPEARANCE_UI_IGNORE_ALPHA | KEEP_APART
 	mouse_opacity = 0
 
-/// Flicks a emote over a mob for a determinable amount of time. Can only be used on mob/living.
+/**
+ * Flick_emote_popup_on_(???) -- A proc type that, when called, causes a image/sprite to appear above whatever entity it is called on.
+ *
+ * There are two types: On_mob and on_obj, they can only be called on their respective typepaths.
+ *
+ * Arguments:
+ * * state -- The ID of whatever .dmi file you're attempting to use for the sprite, in "" format. Ex. "combat", not combat.dmi.
```

Comment:
I think you meant the icon_state? Because looking at the code, that's what it looks to me.